src/Hopper/Internal/LoweredCore/EvalClosureConvertedANF.hs

```diff
@@ -531,6 +533,14 @@ enterTotalMathPrimopSimple controlstack (opId, refs) = do
         argAsLiteral (ValueLitCC lit) = Right lit
         argAsLiteral notLit = Left $ "Not a literal: `" ++ show notLit ++ "`"
 
+enterCaseCC
+  :: SymbolRegistryCC
+  -> EnvStackCC
+  -> ControlStackCC
+  -> (Variable, [(ConstrId, Word32, V.Vector BinderInfoCC, AnfCC)])
```

Comment:
it might be useful to think about having distinct newtype or data type for each of these packing / unpacking pairs, because this seems pretty tuply-typed, which is honestly fine, but always nerve wracking when you want to try to remember / document what is what